### PR TITLE
fallback support factoryBean

### PR DESCRIPTION
**Pull request by min1854**
### Describe what this PR does / why we need it
fallback support factoryBean

### Does this pull request fix one issue?

https://github.com/spring-cloud/spring-cloud-openfeign/issues/822

### Describe how to verify it
junit test

### CODE REVIEW

**Comment by min1854** [comment]
I can build normally locally

**Comment by min1854** [comment]
I didn't understand the issue, I run it locally as normal

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/FallbackSupportFactoryBeanTests.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author 黄学敏（huangxuemin)
+ */
+public class FallbackSupportFactoryBeanTests {
```
Comment (ruansheng8) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/FallbackSupportFactoryBeanTests.java]: The class name should start with `Sentinel` , For example, `SentinelFallbackSupportFactoryBeanTests`

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/feign/SentinelFeign.java [hunk 2]
```diff
@@ -138,6 +140,16 @@ private Object getFromContext(String name, String type,
 								type, fallbackType, name));
 					}
```
Comment (ruansheng8) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/feign/SentinelFeign.java]: Add a comment here to explain what issue was fixed, e.g. fix #3135.

**Review by ruansheng8 — CHANGES_REQUESTED** [review]

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/SentinelFallbackSupportFactoryBeanTests.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	private static class OriginalFeignFallback implements OriginalFeign {
+
+		@Override
+		public String get() {
+			return ORIGINAL_FALLBACK_MESSAGE;
+		}
+
+	}
+
+}
```
Comment (ruansheng8) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/SentinelFallbackSupportFactoryBeanTests.java:101]: The `CheckStyle` plugin is temporarily unavailable on the `2022.x` branch. Please copy the PR code to the `2021.x` branch and then run `mvn clean validate` to check the code. For example, each file should end with a blank line.
See https://github.com/spring-cloud/spring-cloud-build#checkstyle

image

**Review by ruansheng8 — CHANGES_REQUESTED** [review]

**Comment by ruansheng8** [comment]
The `CheckStyle` plugin is temporarily unavailable on the `2022.x` branch. Please copy the PR code to the `2021.x` branch and then run `mvn clean validate` to check the code. For example, each file should end with a blank line.
See https://github.com/spring-cloud/spring-cloud-build#checkstyle

image

**Review by ruansheng8 — APPROVED** [review]

**Review by steverao — APPROVED** [review]
Please check whether need to create a same PR to branch of 2021.x and 2.2.x. They are also main branch of community.

**Comment by min1854** [comment]
> Please check whether need to create a same PR to branch of 2021.x and 2.2.x. They are also main branch of community.

Does it mean me? If so, how should I do this, create two pull requests?

**Comment by ruansheng8** [comment]
> > Please check whether need to create a same PR to branch of 2021.x and 2.2.x. They are also main branch of community.
> 
> Does it mean me? If so, how should I do this, create two pull requests?

Yes , the changes in this PR are applicable to both the `2021.x` and `2.2.x` branches, please submit two separate PR to merge them into `2021.x` and `2.2.x`, respectively.

**Pull request merged**